src/main/java/com/minecolonies/util/BlockUtils.java

```diff
@@ -106,4 +107,132 @@ public static boolean isWater(IBlockState iBlockState)
         return Objects.equals(iBlockState, Blocks.water.getDefaultState())
                || Objects.equals(iBlockState, Blocks.flowing_water.getDefaultState());
     }
+
+    /**
+     * Gets an item from a block.
+     * @param block input block.
+     * @return output item.
+     */
+    public static Item getItemFromBlock(Block block)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This method has 121 lines, which is greater than the 100 lines authorized. Split it into smaller methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS138)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this method 29, down to the maximum allowed 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1142)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "getItemFromBlock" is 66 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)